src/gevent_server.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""Gevent implemnation of http server."""
+from server import response_error
+from server import response_ok
+from server import resolve_uri
+from server import parse_request
+
+
+def server(conn, address):
+    """Return message to client."""
+    try:
+        buffer_length = 8
+        reply_complete = False
+        full_string = u""
+        while not reply_complete:
+            part = conn.recv(buffer_length)
+            full_string = full_string + part.decode('utf-8')
```

Comment:
There's a non-trivial chance that decoding partial messages like this will end up catching a multi-byte unicode character right in the middle.  That will be problematic.